examples/pennylane_run_rotoselect.py

```diff
@@ -0,0 +1,432 @@
+# one wishes to minimize a cost function with respect to some parameters in the circuit. It is desirable
+# to keep the circuit as shallow as possible to reduce the effects of noise, but a given
+# choice of gates is generally suboptimal for performing the optimization.
+# Therefore it would be useful to employ an
+# algorithm which learns a good circuit structure at fixed depth to minimize the cost function.
+#
+# Furthermore, PennyLane's optimizers perform automatic differentiation of quantum nodes by evaluating phase-shifted
+# expectation values using the quantum circuit itself.
+# The output of these calculations, the gradient, is used in optimization methods to minimize
+# the cost function. However,
+# there exists a technique to discover the optimal parameters of a quantum circuit through phase-shifted evaluations,
+# without the need for calculating the gradient as an intermediate step (i.e., a gradient-free optimization).
+# It could be desirable, in some cases, to
+# take advantage of this.
+#
+#
+# The Rotoselect algorithm addresses the above two points: it allows on to jump directly to the
+# optimal value for a single parameter
+# with respect to fixed values for the other parameters, skipping gradient descent, and tries various
+# rotation gates along the way.
+# The algorithm works by updating the parameters :math:`\boldsymbol{\theta}=\theta_1,\dots,\theta_D` and gate choices
+# :math:`\boldsymbol{P}=P_1...P_D`
+# one at a time according to a *closed-form expression for the optimal parameter value* :math:`\theta^{*}_d`
+# when the other parameters and gate choices are fixed:
+#
+# .. math::
+#
+#   \theta^{*}_d &= \underset{\theta_d}{\text{argmin}} \langle H \rangle_{\theta_d} \\
+#                &= -\frac{\pi}{2} - \text{arctan}\left(\frac{2\langle H \rangle_{\theta_d = 0} -
+#                \langle H \rangle_{\theta_d=\pi/2} - \langle H \rangle_{\theta_d=-\pi/2}}{\langle
+#                H \rangle_{\theta_d=\pi/2} -
+#                \langle H \rangle_{\theta_d=-\pi/2}}\right)
+#
+# The calculation makes use of 3 separate evaluations
+# of the expectation value :math:`\langle H \rangle_{\theta_d}` using the quantum circuit. Although
+# :math:`\langle H \rangle` is really a function of all parameters and gate choices
+# :math:`\boldsymbol{\theta}`, :math:`\boldsymbol{P}`, we
+# are fixing every parameter and gate choice apart from :math:`\theta_d` in this expression so we write it as
+# :math:`\langle H \rangle = \langle H \rangle_{\theta_d}`.
+# For each parameter in the quantum circuit, the algorithm proceeds by evaluating :math:`\theta^{*}_d`
+# for each choice of
+# gate :math:`R_d \in \{R_x,R_y,R_z\}` and selecting the gate which yields the minimum value of
+# :math:`\langle H \rangle`.
+#
+# Thus, one might expect the number of circuit evaluations required to be 9 for each parameter (3 for each gate
+# choice). However, since there is a 3-fold
+# degeneracy in the expectation value when :math:`\theta_d=0`,
```

Comment:
```suggestion
# degeneracy in the gates when :math:`\theta_d=0`,
```

Reply:
Both are true -- degeneracy in gates leads to that for expectation -- but the expectations are the evaluations we are concerned with to calculate the optimal parameter values. I'm going to leave this as is for now.

Reply:
Sure, but in the line directly below, you write the degeneracy in terms of the gates.

Reply:
Okay, going with the first suggestion on second thought, thanks.